### PR TITLE
support embedded kubelet config via YAML and JSON

### PR DESCRIPTION
**Pull request by ndbaker1**
**Issue #, if available:**

**Description of changes:**

rather than take an inline config, represented as a `string`, use documents embedded into the `NodeConfig` yaml in any supported format (YAML/JSON).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

File: nodeadm/api/v1alpha1/nodeconfig_types.go [hunk 1]
```diff
@@ -43,10 +44,10 @@ type ClusterDetails struct {
 }
 
 type KubeletOptions struct {
-	// Config is a raw document of a kubelet config that can be provided
-	// by the user to override default generated configurations
+	// Config is a kubelet config that can be provided by the user to override
+	// default generated configurations
 	// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1/
-	Config string `json:"config,omitempty"`
+	Config map[string]runtime.RawExtension `json:"config,omitempty"`
```
Comment (ndbaker1) [inline nodeadm/api/v1alpha1/nodeconfig_types.go:50]: `map[string]interface{}` isn't allowed by conversion-gen, so this is on of the next best suggestions to achieve the same end usage. the internal usages are a bit more complex based on instances where u cannot directly merge `map[string]interface{}` with `map[string]runtime.RawExtension`

File: nodeadm/test/e2e/cases/kubelet-config-dir/config.kubelet-config-json.yaml [hunk 2]
```diff
@@ -8,12 +8,11 @@ spec:
...
-    config: |
-      {
-        "kind": "KubeletConfiguration",
-        "apiVersion": "kubelet.config.k8s.io/v1beta1",
-        "clusterDNS": [
-          "0.0.0.0",
-          "1.1.1.1"
-        ]
-      }
+    config: {
```
Comment (jmdeal) [inline nodeadm/test/e2e/cases/kubelet-config-dir/config.kubelet-config-json.yaml:11]: Is this an intended use case, embedding json inside the yaml? My impression was that users could supply the **entire** `NodeConfig` as either json or yaml, not a subset. I didn't realize this was even valid yaml to begin with but if the tests run and parse I guess it is.

Comment (ndbaker1) [reply nodeadm/test/e2e/cases/kubelet-config-dir/config.kubelet-config-json.yaml:11]: for drop-in use cases I can see users doing it for the `kubelet.config` even though converting it wouldn't take much effort

Comment (cartermckinnon) [reply nodeadm/test/e2e/cases/kubelet-config-dir/config.kubelet-config-json.yaml:11]: All JSON is valid YAML, so I guess this is allowed per the specs. Looks pretty odd though 😄

File: nodeadm/api/v1alpha1/nodeconfig_types.go (hunk 1 above)
Comment (cartermckinnon) [reply nodeadm/api/v1alpha1/nodeconfig_types.go:50]: This looks like the way to go. If merging this type with mergo is cumbersome, can we write a custom handler or something? Marshalling to-and-from JSON will work for now but would be nice to clean that up

Comment (ndbaker1) [reply nodeadm/api/v1alpha1/nodeconfig_types.go:50]: sounds good 👍, as long as the use case works we can hash this out

**Pull request merged**